File: core/python/src/aiddl_core/function/uri.py
from aiddl_core.representation.symbolic import Symbolic

EVAL = Symbolic("org.aiddl.eval")
EVAL_REF = Symbolic("org.aiddl.eval.eval-ref")
EVAL_ALL_REFS = Symbolic("org.aiddl.eval.eval-all-refs")

FUNCTION_LOADER = Symbolic("org.aiddl.eval.load-function")
FACTORY_LOADER = Symbolic("org.aiddl.eval.load-function-factory")
CORE_LANG = Symbolic("org.aiddl.eval.core-lang")

# org.aiddl.eval

CALL = Symbolic("org.aiddl.eval.call")
CALL_REQUEST = Symbolic("org.aiddl.eval.call-request")
LAMBDA = Symbolic("org.aiddl.eval.lambda")
QUOTE = Symbolic("org.aiddl.eval.quote")
TYPE = Symbolic("org.aiddl.eval.type")
ZIP = Symbolic("org.aiddl.eval.zip")
MAP = Symbolic("org.aiddl.eval.map")
FILTER = Symbolic("org.aiddl.eval.filter")
REDUCE = Symbolic("org.aiddl.eval.reduce")
DOMAIN = Symbolic("org.aiddl.eval.domain")
EVAL_REF = Symbolic("org.aiddl.eval.eval-ref")
SIGNATURE = Symbolic("org.aiddl.eval.signature")
MATCHES = Symbolic("org.aiddl.eval.matches")
EQUALS = Symbolic("org.aiddl.eval.equals")
NOT_EQUALS = Symbolic("org.aiddl.eval.not-equals")
SUBSTITUTE = Symbolic("org.aiddl.eval.substitute")
GET_MATCHING_ENTRIES = Symbolic("org.aiddl.eval.get-matching-entries")

FIRST = Symbolic("org.aiddl.eval.first")
LAST = Symbolic("org.aiddl.eval.last")
KEY = Symbolic("org.aiddl.eval.key")
VALUE = Symbolic("org.aiddl.eval.value")
SIZE = Symbolic("org.aiddl.eval.size")
GET_IDX = Symbolic("org.aiddl.eval.get-idx")
GET_KEY = Symbolic("org.aiddl.eval.get-key")
CONTAINS_KEY = Symbolic("org.aiddl.eval.contains-key")
IF = Symbolic("org.aiddl.eval.if")
COND = Symbolic("org.aiddl.eval.cond")
MATCH = Symbolic("org.aiddl.eval.match")
LET = Symbolic("org.aiddl.eval.let")

# org.aiddl.eval.symbolic

SYM_CONCAT = Symbolic("org.aiddl.eval.symbolic.concat")
SYM_SPLIT = Symbolic("org.aiddl.eval.symbolic.split")

# org.aiddl.eval.string

STR_CONCAT = Symbolic("org.aiddl.eval.string.concat")
STR_MOD_NAME = Symbolic("org.aiddl.eval.string.module-folder")

# org.aiddl.eval.collection

IN = Symbolic("org.aiddl.eval.collection.in")
CONTAINS = Symbolic("org.aiddl.eval.collection.contains")
CONTAINS_ALL = Symbolic("org.aiddl.eval.collection.contains-all")
CONTAINS_ANY = Symbolic("org.aiddl.eval.collection.contains-any")
ADD_COL = Symbolic("org.aiddl.eval.collection.add-element")
ADD_COL_ALL = Symbolic("org.aiddl.eval.collection.add-all")
REM_COL = Symbolic("org.aiddl.eval.collection.remove")
REM_COL_ALL = Symbolic("org.aiddl.eval.collection.remove-all")
CONTAINS_MATCH = Symbolic("org.aiddl.eval.collection.contains-match")
PUT_ALL = Symbolic("org.aiddl.eval.collection.put-all")
SUM = Symbolic("org.aiddl.eval.collection.sum")

# org.aiddl.eval.set

UNION = Symbolic("org.aiddl.eval.set.union")

# org.aiddl.eval.list

CONCAT = Symbolic("org.aiddl.eval.list.concat")

# org.aiddl.eval.numeric

ADD = Symbolic("org.aiddl.eval.numerical.add")
SUB = Symbolic("org.aiddl.eval.numerical.sub")
MULT = Symbolic("org.aiddl.eval.numerical.mult")
DIV = Symbolic("org.aiddl.eval.numerical.div")
MODULO = Symbolic("org.aiddl.eval.numerical.modulo")
GREATER_THAN = Symbolic("org.aiddl.eval.numerical.greater-than")
GREATER_THAN_EQ = Symbolic("org.aiddl.eval.numerical.greater-than-eq")
LESS_THAN = Symbolic("org.aiddl.eval.numerical.less-than")
LESS_THAN_EQ = Symbolic("org.aiddl.eval.numerical.less-than-eq")
IS_POS = Symbolic("org.aiddl.eval.numerical.is-positive")
IS_NEG = Symbolic("org.aiddl.eval.numerical.is-negative")
IS_ZERO = Symbolic("org.aiddl.eval.numerical.is-zero")
IS_INF = Symbolic("org.aiddl.eval.numerical.is-infinite")
IS_INF_POS = Symbolic("org.aiddl.eval.numerical.is-infinite-positive")
IS_INF_NEG = Symbolic("org.aiddl.eval.numerical.is-infinite-negative")
IS_NAN = Symbolic("org.aiddl.eval.numerical.is-nan")

# org.aiddl.eval.random

UNIFORM_RANDOM = Symbolic("org.aiddl.eval.random.uniform.real")
UNIFORM_RANDOM_ELEMENT = Symbolic("org.aiddl.eval.random.uniform.element")
UNIFORM_RANDOM_INTEGER = Symbolic("org.aiddl.eval.random.uniform.integer")
NORMAL_RANDOM = Symbolic("org.aiddl.eval.random.normal")


# org.aiddl.eval.logic

AND = Symbolic("org.aiddl.eval.logic.and")
OR = Symbolic("org.aiddl.eval.logic.or")
NOT = Symbolic("org.aiddl.eval.logic.not")
EXISTS = Symbolic("org.aiddl.eval.logic.exists")
FORALL = Symbolic("org.aiddl.eval.logic.forall")

# org.aiddl.type

TYPE_TERM = Symbolic("org.aiddl.type.term")
TYPE_NUMERICAL = Symbolic("org.aiddl.type.term.numerical")
TYPE_INTEGER = Symbolic("org.aiddl.type.term.numerical.integer")
TYPE_RATIONAL = Symbolic("org.aiddl.type.term.numerical.rational")
TYPE_REAL = Symbolic("org.aiddl.type.term.numerical.real")
TYPE_INF = Symbolic("org.aiddl.type.term.numerical.infinity")
TYPE_SYMBOLIC = Symbolic("org.aiddl.type.term.symbolic")
TYPE_BOOLEAN = Symbolic("org.aiddl.type.term.symbolic.boolean")
TYPE_STRING = Symbolic("org.aiddl.type.term.string")
TYPE_VARIABLE = Symbolic("org.aiddl.type.term.variable")
TYPE_REF = Symbolic("org.aiddl.type.term.reference")
TYPE_FREF = Symbolic("org.aiddl.type.term.fun-ref")
TYPE_COLLECTION = Symbolic("org.aiddl.type.term.collection")
TYPE_SET = Symbolic("org.aiddl.type.term.collection.set")
TYPE_LIST = Symbolic("org.aiddl.type.term.collection.list")
TYPE_TUPLE = Symbolic("org.aiddl.type.term.tuple")
TYPE_KEY_VALUE = Symbolic("org.aiddl.type.term.key-value")

File: core/java/src/main/java/org/aiddl/core/function/Uri.java
package org.aiddl.core.function;

import org.aiddl.core.representation.SymbolicTerm;
import org.aiddl.core.representation.Term;

public class Uri {
    public static final SymbolicTerm EVAL = Term.sym("org.aiddl.eval");
    public static final SymbolicTerm EVAL_REF = Term.sym("org.aiddl.eval.eval-ref");
    public static final SymbolicTerm EVAL_ALL_REFS = Term.sym("org.aiddl.eval.eval-all-refs");

    public static final SymbolicTerm LOAD_JAVA = Term.sym("org.aiddl.eval.load-function");
    public static final SymbolicTerm LOAD_JAVA_FACT = Term.sym("org.aiddl.eval.load-function-factory");

    public static final SymbolicTerm CORE_LANG = Term.sym("org.aiddl.eval.core-lang");


    public static final SymbolicTerm INIT_FUNCTION = Term.sym("org.aiddl.eval.init-function");
    public static final SymbolicTerm CONFIG_FUNCTION = Term.sym("org.aiddl.eval.config-function");

    /**
     * org.aiddl.eval
     */
    public static final SymbolicTerm CALL = Term.sym("org.aiddl.eval.call");
    public static final SymbolicTerm LAMBDA = Term.sym("org.aiddl.eval.lambda");
    public static final SymbolicTerm QUOTE = Term.sym("org.aiddl.eval.quote");
    public static final SymbolicTerm TYPE = Term.sym("org.aiddl.eval.type");
    public static final SymbolicTerm ZIP = Term.sym("org.aiddl.eval.zip");
    public static final SymbolicTerm MAP = Term.sym("org.aiddl.eval.map");
    public static final SymbolicTerm FILTER = Term.sym("org.aiddl.eval.filter");
    public static final SymbolicTerm REDUCE = Term.sym("org.aiddl.eval.reduce");
    public static final SymbolicTerm DOMAIN = Term.sym("org.aiddl.eval.domain");
    public static final SymbolicTerm SIGNATURE = Term.sym("org.aiddl.eval.signature");
    public static final SymbolicTerm MATCHES = Term.sym("org.aiddl.eval.matches");
    public static final SymbolicTerm EQUALS = Term.sym("org.aiddl.eval.equals");
    public static final SymbolicTerm NOT_EQUALS = Term.sym("org.aiddl.eval.not-equals");
    public static final SymbolicTerm SUBSTITUTE = Term.sym("org.aiddl.eval.substitute");
    public static final SymbolicTerm GET_MATCHING_ENTRIES = Term.sym("org.aiddl.eval.get-matching-entries");

    public static final SymbolicTerm FIRST = Term.sym("org.aiddl.eval.first");
    public static final SymbolicTerm LAST = Term.sym("org.aiddl.eval.last");
    public static final SymbolicTerm KEY = Term.sym("org.aiddl.eval.key");
    public static final SymbolicTerm VALUE = Term.sym("org.aiddl.eval.value");
    public static final SymbolicTerm SIZE = Term.sym("org.aiddl.eval.size");
    public static final SymbolicTerm GET_IDX = Term.sym("org.aiddl.eval.get-idx");
    public static final SymbolicTerm GET_KEY = Term.sym("org.aiddl.eval.get-key");
    public static final SymbolicTerm IF = Term.sym("org.aiddl.eval.if");
    public static final SymbolicTerm COND = Term.sym("org.aiddl.eval.cond");
    public static final SymbolicTerm MATCH  = Term.sym("org.aiddl.eval.match");
    public static final SymbolicTerm LET    = Term.sym("org.aiddl.eval.let");

    /**
     * org.aiddl.eval.symbolic
     */
    public static final SymbolicTerm SYM_CONCAT    = Term.sym("org.aiddl.eval.symbolic.concat");
    public static final SymbolicTerm SYM_SPLIT    = Term.sym("org.aiddl.eval.symbolic.split");

    /**
     * org.aiddl.eval.string
     */
    public static final SymbolicTerm STR_CONCAT    = Term.sym("org.aiddl.eval.string.concat");
    public static final SymbolicTerm STR_MOD_NAME    = Term.sym("org.aiddl.eval.string.module-folder");

    /**
     * org.aiddl.eval.collection
     */
    public static final SymbolicTerm IN = Term.sym("org.aiddl.eval.collection.in");
    public static final SymbolicTerm CONTAINS = Term.sym("org.aiddl.eval.collection.contains");
    public static final SymbolicTerm CONTAINS_ALL = Term.sym("org.aiddl.eval.collection.contains-all");
    public static final SymbolicTerm CONTAINS_ANY = Term.sym("org.aiddl.eval.collection.contains-any");
    public static final SymbolicTerm ADD_COL = Term.sym("org.aiddl.eval.collection.add-element");
    public static final SymbolicTerm ADD_COL_ALL = Term.sym("org.aiddl.eval.collection.add-all");
    public static final SymbolicTerm REM_COL = Term.sym("org.aiddl.eval.collection.remove");
    public static final SymbolicTerm REM_COL_ALL = Term.sym("org.aiddl.eval.collection.remove-all");
    public static final SymbolicTerm CONTAINS_MATCH = Term.sym("org.aiddl.eval.collection.contains-match");
    public static final SymbolicTerm SUM = Term.sym("org.aiddl.eval.collection.sum");
    /**
     * collection & tuple
     */
    public static final SymbolicTerm PUT_ALL = Term.sym("org.aiddl.eval.collection.put-all");
    public static final SymbolicTerm CONTAINS_KEY = Term.sym("org.aiddl.eval.contains-key");
    public static final SymbolicTerm IS_UNIQUE_MAP = Term.sym("org.aiddl.eval.is-unique-map");
    /**
     * org.aiddl.eval.set
     */
    public static final SymbolicTerm UNION	= Term.sym("org.aiddl.eval.set.union");
    /**
     * org.aiddl.eval.list
     */
    public static final SymbolicTerm CONCAT	= Term.sym("org.aiddl.eval.list.concat");
    /**
     * org.aiddl.eval.numeric
     */
    public static final SymbolicTerm ADD = Term.sym("org.aiddl.eval.numerical.add");
    public static final SymbolicTerm SUB = Term.sym("org.aiddl.eval.numerical.sub");
    public static final SymbolicTerm MULT = Term.sym("org.aiddl.eval.numerical.mult");
    public static final SymbolicTerm DIV = Term.sym("org.aiddl.eval.numerical.div");
    public static final SymbolicTerm MODULO = Term.sym("org.aiddl.eval.numerical.modulo");

    public static final SymbolicTerm GREATER_THAN = Term.sym("org.aiddl.eval.numerical.greater-than");
    public static final SymbolicTerm GREATER_THAN_EQ = Term.sym("org.aiddl.eval.numerical.greater-than-eq");
    public static final SymbolicTerm LESS_THAN = Term.sym("org.aiddl.eval.numerical.less-than");
    public static final SymbolicTerm LESS_THAN_EQ = Term.sym("org.aiddl.eval.numerical.less-than-eq");

    public static final SymbolicTerm IS_POSITIVE = Term.sym("org.aiddl.eval.numerical.is-negative");
    public static final SymbolicTerm IS_NEGATIVE = Term.sym("org.aiddl.eval.numerical.is-zero");
    public static final SymbolicTerm IS_ZERO = Term.sym("org.aiddl.eval.numerical.is-zero");
    public static final SymbolicTerm IS_NAN = Term.sym("org.aiddl.eval.numerical.is-nan");

    /**
     * org.aiddl.eval.logic
     */
    public static final SymbolicTerm AND = Term.sym("org.aiddl.eval.logic.and");
    public static final SymbolicTerm OR  = Term.sym("org.aiddl.eval.logic.or");
    public static final SymbolicTerm NOT = Term.sym("org.aiddl.eval.logic.not");
    public static final SymbolicTerm EXISTS = Term.sym("org.aiddl.eval.logic.exists");
    public static final SymbolicTerm FORALL = Term.sym("org.aiddl.eval.logic.forall");

    /**
     * org.aiddl.type
     */
    public static final SymbolicTerm TYPE_TERM = Term.sym("org.aiddl.type.term");
    public static final SymbolicTerm TYPE_NUMERICAL = Term.sym("org.aiddl.type.term.numerical");
    public static final SymbolicTerm TYPE_INTEGER = Term.sym("org.aiddl.type.term.numerical.integer");
    public static final SymbolicTerm TYPE_RATIONAL = Term.sym("org.aiddl.type.term.numerical.rational");
    public static final SymbolicTerm TYPE_REAL = Term.sym("org.aiddl.type.term.numerical.real");
    public static final SymbolicTerm TYPE_INF = Term.sym("org.aiddl.type.term.numerical.infinity");

    public static final SymbolicTerm TYPE_SYMBOLIC = Term.sym("org.aiddl.type.term.symbolic");
    public static final SymbolicTerm TYPE_BOOLEAN = Term.sym("org.aiddl.type.term.symbolic.boolean");
    public static final SymbolicTerm TYPE_STRING = Term.sym("org.aiddl.type.term.string");
    public static final SymbolicTerm TYPE_VARIABLE = Term.sym("org.aiddl.type.term.variable");
    public static final SymbolicTerm TYPE_REF = Term.sym("org.aiddl.type.term.reference");
    public static final SymbolicTerm TYPE_FREF = Term.sym("org.aiddl.type.term.fun-ref");

    public static final SymbolicTerm TYPE_COLLECTION = Term.sym("org.aiddl.type.term.collection");
    public static final SymbolicTerm TYPE_SET = Term.sym("org.aiddl.type.term.collection.set");
    public static final SymbolicTerm TYPE_LIST = Term.sym("org.aiddl.type.term.collection.list");
    public static final SymbolicTerm TYPE_TUPLE = Term.sym("org.aiddl.type.term.tuple");
    public static final SymbolicTerm TYPE_KEY_VALUE = Term.sym("org.aiddl.type.term.key-value");
}
